### PR TITLE
add early bailout

### LINKED ISSUE #26: Early bail out
@poteto is there a easy way to prevent the mixin from beeing initialized? Lets say i set `viewportEntered = true` and `_setupElement` would bail out immediately. That would be great for `ember-lazy-image` when the image is already in cache and would save a lot of performance.

### PR DESCRIPTION
**Pull request by fsmanuel**
@poteto should it be more explicit e.g. `noSetup =  true` or is `viewportEntered = true` sufficient?

closes #26

### CODE REVIEW

**Comment by poteto** [comment]
I think it'll be best to have a new prop called `noSetup`, and leave `viewportEntered` read only

**Comment by fsmanuel** [comment]
@poteto updated to `noSetup`

File: addon/mixins/in-viewport.js [hunk 1]
```diff
@@ -59,7 +59,9 @@ export default Ember.Mixin.create({
   },
 
   _setupElement: on('didInsertElement', function() {
-    if (!canUseDOM) {
+    const noSetup = !get(this, 'viewportSpy') && get(this, 'noSetup');
```
Comment (kellyselden) [inline addon/mixins/in-viewport.js]: I would suggest ignoring `viewportSpy` here. When someone wants to disable the component, they shouldn't have to worry about the other option values.

**Comment by kellyselden** [comment]
@fsmanuel @poteto Offered a comment to help get this merged.

File: addon/mixins/in-viewport.js (hunk 1 above)
Comment (fsmanuel) [inline addon/mixins/in-viewport.js]: @kellyselden Agreed! @poteto should I rebase and update according to kellyselden's suggestion?

**Comment by kellyselden** [comment]
I think `develop` is a dead branch. You should branch off of and PR into the `master` branch.

**Comment by fsmanuel** [comment]
@kellyselden thanks didn't know that develop is dead 👍 

closing in favor of #90

**Pull request closed without merging**